ResearchSuite/ResearchSuite/RSDJSONValue.swift

```diff
@@ -185,12 +185,12 @@ extension DateComponents : RSDJSONValue {
                 formatString = "yyyy"
             }
             if let month = month, month > 0 {
-                if formatString.characters.count > 0 {
+                if formatString.count > 0 {
                     formatString.append("-")
                 }
                 formatString.append("MM")
                 if let day = day, day > 0 {
-                    if formatString.characters.count > 0 {
+                    if formatString.count > 0 {
```

Comment:
won't it always be >0 anyway since you just appended "MM" above?

Reply:
doesn't matter. looking for the last match below this so if there twice, its the last trip that matters.